doc/api/esm.md

```diff
@@ -1206,6 +1206,9 @@ _defaultEnv_ is the conditional environment name priority array,
 > 1. If _packageSubpath_ contains any _"."_ or _".."_ segments or percent
 >    encoded strings for _"/"_ or _"\\"_, then
 >    1. Throw an _Invalid Specifier_ error.
+> 1. Set _selfUrl_ to the result of
```

Comment:
Does this change mean that self-resolve works different in `import` than in `require`? E.g. the following will potentially load different packages?

```js
import('foo'); // will prefer self-ref
require('foo'); // will prefer node_modules/foo
```

That feels awkward. Is there a real advantage of changing the order here?

Reply:
Yes. The advantage is that self-resolve is encapsulated to be unaffected by node_modules installs. I think this encapsulation is important for the meaning of self resolution.

Reply:
Marking as resolved with the latest changes.